tests/test_library_conf.py

```diff
@@ -62,5 +62,121 @@ def test_trace_header_tags(self):
         tag_config_list = full_tag_config_list[1::]
 
         tags = {item.split(":")[1]: self.headers[item.split(":")[0]] for item in tag_config_list}
+        interfaces.library.add_span_tag_validation(request=self.r, tags=tags)
+
+
+@irrelevant(library="cpp")
+@released(dotnet="2.1.0", golang="1.53.0", java="0.102.0", nodejs="?", php="0.74.0", python="1.2.1", ruby="?")
+@coverage.basic
+@scenarios.library_conf_custom_headers_whitespacing_headers
+class Test_HeaderTagsWhitespacing_Headers:
+    """Validates that leading/trailing whitespaces are trimmed on the header values given to DD_TRACE_HEADER_TAGS
+    e.g, ' header ' in DD_TRACE_HEADER_TAGS=' header ' becomes 'header' and is expected to match req.header of 'header' """
+
+    def setup_trace_header_tags(self):
+        self.headers = {"header-tag1": "header-val1"}
+        self.r = weblog.get("/waf", headers=self.headers)
+
+    def test_trace_header_tags(self):
+        tag_conf = scenarios.library_conf_custom_headers_whitespacing_headers.weblog_container.environment[
+            "DD_TRACE_HEADER_TAGS"
+        ]
+
+        full_tag_config_list = tag_conf.split(",")
+        # skip the first item, as this required to make the tests work on some platforms
+        tag_config_list = full_tag_config_list[1::]
+
+        tags = {}
+        for tag in tag_config_list:
+            tag = tag.strip()
+            tags["http.request.headers." + tag] = re.sub("[^a-zA-Z0-9 -]", "_", self.headers[tag])
+
+        interfaces.library.add_span_tag_validation(request=self.r, tags=tags)
+
+
+# Note: dotnet fails this test. It normalizes those spaces to underscores instead.
```

Comment:
so you must decorate the class with `@bug(library="dotnet", reason="It normalizes those spaces to underscores instead.")`

Cherry on the cake, the Jira ticket as the reason

Reply:
in progress: https://dd.slack.com/archives/CC7Q58YRG/p1694114623619059

Reply:
I created this ticket for us to look into this: https://datadoghq.atlassian.net/browse/AIT-8308
Thanks!